src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -441,6 +441,63 @@ TEST(Aktualizr, FullWithUpdatesNeedReboot) {
   }
 }
 
+/*
+ * Initialize -> UptaneCycle -> updates downloaded and installed for primary
+ * -> reboot emulated -> Initialize -> Finalize Installation After Reboot
+ *
+ * Verifies an auto reboot and pending updates 'finalization' after a reboot
+ *
+ * Checks actions:
+ *
+ * - [x] Auto reboot triggering just after installation if `force_install_completion` is set
+ *   - [x] Update download and initial installation result storing
+ *   - [x] Exit from Uptane Cycle just after installation
+ *   - [x] Reboot just after installation and exit form the uptane cycle
+ *
+ *  - [x] Pending updates are installed on primary after an auto-rebbot
+ */
+TEST(Aktualizr, AutoRebootAfterUpdate) {
+  TemporaryDirectory temp_dir;
+  auto http = std::make_shared<HttpFakePutCounter>(temp_dir.Path());
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, http->tls_server);
+  conf.pacman.fake_need_reboot = true;
+  conf.uptane.force_install_completion = true;
+  conf.uptane.polling_sec = 0;
+  conf.bootloader.reboot_sentinel_dir = temp_dir.Path();
+
+  {
+    // first run: do the install, exit UptaneCycle and emulate reboot since force_install_completion is set
+    auto storage = INvStorage::newStorage(conf.storage);
+    Aktualizr aktualizr(conf, storage, http);
+
+    aktualizr.Initialize();
+    auto aktualizr_cycle_thread = aktualizr.RunForever();
+    auto aktualizr_cycle_thread_status = aktualizr_cycle_thread.wait_for(std::chrono::seconds(4));
```

Comment:
So the 4 seconds is a timeout and shouldn't be reached, I think?

Maybe set a larger value so that it won't fail even on loaded CI machines, maybe 20 seconds?